lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelResult.cxx

```diff
@@ -245,6 +245,25 @@ Point LinearModelResult::getCoefficientsStandardErrors() const
   return standardErrors;
 }
 
+Matrix LinearModelResult::getDesign() const
+{
+  return design_;
+}
+
+SymmetricMatrix LinearModelResult::computeProjectionMatrix() const
+{
+  SymmetricMatrix gramMatrix(computeGramMatrix());
+  SymmetricMatrix inverseGramMatrix(gramMatrix.inverse());
+  Matrix projection(design_.transpose() * inverseGramMatrix * design_);
+  return SymmetricMatrix(projection.getImplementation());
+}
+
+SymmetricMatrix LinearModelResult::computeGramMatrix() const
+{
+  Matrix gram(design_.transpose() * design_);
+  return SymmetricMatrix(gram.getImplementation());
```

Comment:
No need to cast the result here. IMO the whole method should be removed as the MatrixImplementation::computeGram() method do exactly the job.

Reply:
Good point: fixed at 1a9d8fbce.